sky/services/ns_net/url_loader_impl.mm

```diff
@@ -104,12 +166,27 @@ - (void)dealloc {
 
 void URLLoaderImpl::Start(URLRequestPtr request,
                           const StartCallback& callback) {
+  if (request->body.size() == 1) {
+    // If the body has request data, try to drain that
+    AsyncNSDataDrainer::Drain(
+        request->body[0].Pass(),  // handle
+        base::Bind(&URLLoaderImpl::StartNow, base::Unretained(this),
```

Comment:
This lifetimes here are slightly sketchy.  In particular, `this` can die before the drain is finished and you'll get a callback call with a bogus this pointer.  You've got two choices for how to resolve this issue:

1. Have this object hold the AsyncNSDataDrainer as a member, similar to how AsyncNSDataDrainer holds common::DataPipeDrainer.  Then when this object dies, it will kill the AsyncNSDataDrainer, which will kill the common::DataPipeDrainer and stop the callbacks.

2. Instead of using `base::Unretained(this)` here, you can bind a `base::WeakPtr` to the callback.  The callback will check the weak pointer before using it as the "this" pointer.

I'd recommend (1) because the WeakPtr approach doesn't stop the drain when the client goes away.